### PR TITLE
Hotfix for failure with newest xarray version

### PR DESCRIPTION
**Pull request by jbusecke**
- [x] Passes `isort . && black . && flake8`
 - [x] Fully documented, including `whats-new.rst`

I have noticed that the CI over at https://github.com/jbusecke/cmip6_preprocessing/issues/173 is broken. Some of these changes can be fixed there, but there are some failures that are caused by this package.

This is a rather naive fix for the issue.

### CODE REVIEW

**Comment by jbusecke** [comment]
I think this might also require to drop support for 3.6. @mathause what do you think?

**Comment by jbusecke** [comment]
Eh actually this works out. I dropped 3.6 for the regular matrix CI but kept it for the 'bare-minimum' config. Do you think that is sufficient?

**Comment by mathause** [comment]
Thanks! I hadn't even noticed :see_no_evil: I need to take a look.

Would be nice if py3.6 still worked unconditionally but I can also fix this internally (I am dropping py2.7 in this version and didn't want to drop p3.6 as well).

**Comment by jbusecke** [comment]
I am not quite sure which tests are skipped when using the bare-minimum config. We could also add a specific 3.6 environment, but it seems it would be leading to a bunch of clutter?

Happy to make any changes, but if we could merge this relatively soonish, that would save me a lot of trouble. 

>  I hadn't even noticed 🙈

Me neither haha.

**Comment by mathause** [comment]
> if we could merge this relatively soonish, that would save me a lot of trouble.

Do you need a new release or just an update to master?

**Comment by jbusecke** [comment]
> Do you need a new release or just an update to master?

A release would be ideal. I could install from master, but would have to refactor a bunch of my CI environments.

**Comment by jbusecke** [comment]
Are there other issues/PRs that would block a release of this? In that case Id be happy to implement the upstream master in my CI.

File: docs/whats_new.rst [hunk 1]
```diff
@@ -65,6 +65,7 @@ Bug Fixes
   (:issue:`157`).
 - Move ``_flatten_polygons`` to ``utils`` and raise an error when something else than
   a ``Polygon`` or ``MultiPolygon`` is passed (:pull:`211`).
+- Fix errors with masking due to xarray >=0.19.0 (:pull:`234`). By `Julius Busecke <https://github.com/jbusecke>`_.
```
Comment (mathause) [inline docs/whats_new.rst:68]:
```suggestion
- Fix incompatibility with xarray >=0.19.0 (:pull:`234`). By `Julius Busecke <https://github.com/jbusecke>`_.
```

**Review by mathause — APPROVED** [review]
I just tested this and looks all good. I think I can merge and release today. There are no blocking issues or PRs.

I see now what the problem with xr & py36 is - xarray does not restrict version 0.19 to py>=3.7, but should. As I intend to remove py36 support in the next version I am fine with removing it from the test matrix.

I'd also like to overhaul the creation of the xr.DataArray masks & simplify this, but I haven't gotten around.

**Comment by mathause** [comment]
Thanks a lot @jbusecke

**Comment by jbusecke** [comment]
No thank YOU for the quick turnaround!

**Comment by jbusecke** [comment]
Should I raise an issue about the dependency over at xarray? This problem shows up in almost all of my packages.

**Comment by keewis** [comment]
I'm not sure I follow: https://github.com/pydata/xarray/blob/da99a5664df4f5013c2f6b0e758394bec5e0bc80/setup.cfg#L77 correctly specifies `python_version = >=3.7`? Unless you meant that the feedstock was not updated?

**Comment by jbusecke** [comment]
Oh maybe I misunderstood this line

> I see now what the problem with xr & py36 is - xarray does not restrict version 0.19 to py>=3.7, but should.

But if that is already required, how does a py3.6 environment resolve to xarray >= 0.17.0?

**Comment by mathause** [comment]
The issue is the conda build 0 (xarray 0.19.0 pyhd8ed1ab_0) which allows python 3.6, see:

```bash
conda search --info xarray==0.19 
```

I am not sure there is much we can do (except mark it as broken? https://github.com/conda-forge/admin-requests)

**Comment by jbusecke** [comment]
Ah ok, thanks for the clarification. I am unsure about those things, but it might be worth checking if that option might be useful?

**Comment by keewis** [comment]
I think that's it, see conda-forge/xarray-feedstock#66. The first version that officially dropped `py3.6` was `0.17`, but `0.19` appears to be the first incompatible version.

**Pull request merged**